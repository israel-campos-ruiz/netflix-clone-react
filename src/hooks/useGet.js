import { useEffect, useRef, useState } from "react";
import axios from "axios";

const useGet = (url) => {
  const isMounted = useRef(true);
  const [state, setState] = useState({
    data: null,
    loading: true,
    error: null,
  });
  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);
  useEffect(() => {
    const getData = async () => {
      try {
          setState({data:null, loading:true, error:null});
          const response = await axios.get(url);
          const { data } = response;
          setState({data, loading:false, error:null});
      } catch (error) {
          setState({data:null, loading:false, error:'error al cargar la data'}) 
      }
    };
    if(isMounted.current){
        getData();
    }
  }, [url]);

  return state;
};

export default useGet;
