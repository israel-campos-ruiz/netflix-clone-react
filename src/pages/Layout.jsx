import React from "react";
import useGet from "../hooks/useGet";
import RowItem from "../components/layout/RowItem";
import Loader from "../components/shared/Loader";
import { requests } from "../helpers/request";
import Banner from "../components/layout/Banner";
import '../styles/App.css'
import Navbar from "../components/shared/Navbar";

const Layout = () => {
  const { data, loading } = useGet(requests.fetchTrending);
  return (
    <div className="app">
      {loading && <Loader />}
       {/* navbar */}
       <Navbar />
       <Banner/>
      {data &&(
        <div className="app">
          <RowItem isLargeRow title={"NETFLIX ORIGINALS"} fetchUrl={requests.fetchNetflixOriginals}/>
          <RowItem isLargeRow title={"Trending Now"} fetchUrl={requests.fetchTrending}/>
          <RowItem isLargeRow title={"Top Rated"} fetchUrl={requests.fetchTopRated}/>
          <RowItem isLargeRow title={"Action Movies"} fetchUrl={requests.fetchActionMovies}/>
          <RowItem isLargeRow title={"Comedy Movies"} fetchUrl={requests.fetchComedyMovies}/>
          <RowItem isLargeRow title={"Horror Movies"} fetchUrl={requests.fetchHorrorMovies}/>
          <RowItem isLargeRow title={"Romance Movies"} fetchUrl={requests.fetchRomanceMovies}/>
          <RowItem isLargeRow title={"Documentaries"} fetchUrl={requests.fetchDocumentariesMovies}/>
    
        </div>
      )}
    </div>
  );
};

export default Layout;
