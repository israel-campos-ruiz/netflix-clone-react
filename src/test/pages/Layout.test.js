import React from 'react';
import '@testing-library/jest-dom';
import Layout from '../../pages/Layout';
import { shallow } from 'enzyme';
import useGet from '../../hooks/useGet';
jest.mock('../../hooks/useGet');

describe('pruebas en el componente <LayoutScreen />', () => {
    test('Debe de hacer match con el snapshot cuando esta cargando la data con un loader en true', () => {
        useGet.mockReturnValue({
            data:null,
            loading:true,
            error:null
        });
        const wrapper = shallow(<Layout/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Loader').exists()).toBe(true)
    })

    test('Debe de crear un segundo snapshot pero con datos', () => {
        const data =  {
            id: 63174,
            popularity: 572.958,
            vote_count: 6449,
            name: 'Lucifer',
            first_air_date: '2016-01-25',
            backdrop_path: '/ta5oblpMlEcIPIS2YGcq9XEkWK2.jpg',
            overview: "Bored and unhappy as the Lord of Hell, Lucifer Morningstar abandoned his throne and retired to Los Angeles, where he has teamed up with LAPD detective Chloe Decker to take down criminals.Â But the longer he's away from the underworld, the greater the threat that the worst of humanity could escape.",
            genre_ids: [Array],
            original_name: 'Lucifer',
            original_language: 'en',
            vote_average: 8.5,
            poster_path: '/4EYPN5mVIhKLfxGruy7Dy41dTVn.jpg',
            origin_country: [Array]
          }
        useGet.mockReturnValue({
            data,
            loading:false,
            error:null
        })
          const wrapper = shallow(<Layout/>);
          expect(wrapper).toMatchSnapshot()
          expect(wrapper.find('div').at(1).exists()).toBe(true)
    });
   
})
