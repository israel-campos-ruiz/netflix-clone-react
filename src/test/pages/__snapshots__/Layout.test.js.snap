// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <LayoutScreen /> Debe de crear un segundo snapshot pero con datos 1`] = `
<div
  className="app"
>
  <Navbar />
  <Banner />
  <div
    className="app"
  >
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/discover/tv?api_key=f5db8a9255d67e2282b48f43ea95054a&with_networks=213"
      isLargeRow={true}
      title="NETFLIX ORIGINALS"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/trending/all/week?api_key=f5db8a9255d67e2282b48f43ea95054a&language=en-US"
      isLargeRow={true}
      title="Trending Now"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/movie/top_rated?api_key=f5db8a9255d67e2282b48f43ea95054a&language=en-US"
      isLargeRow={true}
      title="Top Rated"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/discover/movie?api_key=f5db8a9255d67e2282b48f43ea95054a&with_genres=28"
      isLargeRow={true}
      title="Action Movies"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/discover/movie?api_key=f5db8a9255d67e2282b48f43ea95054a&with_genres=35"
      isLargeRow={true}
      title="Comedy Movies"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/discover/movie?api_key=f5db8a9255d67e2282b48f43ea95054a&with_genres=27"
      isLargeRow={true}
      title="Horror Movies"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/discover/movie?api_key=f5db8a9255d67e2282b48f43ea95054a&with_genres=10749"
      isLargeRow={true}
      title="Romance Movies"
    />
    <Memo()
      fetchUrl="https://api.themoviedb.org/3/discover/movie?api_key=f5db8a9255d67e2282b48f43ea95054a&with_genres=99"
      isLargeRow={true}
      title="Documentaries"
    />
  </div>
</div>
`;

exports[`pruebas en el componente <LayoutScreen /> Debe de hacer match con el snapshot cuando esta cargando la data con un loader en true 1`] = `
<div
  className="app"
>
  <Loader />
  <Navbar />
  <Banner />
</div>
`;
